### ISSUE TITLE
Permissions from multiple policies not merged during access check?

### ISSUE THREAD

**Issue by dmoagx** [modxcms/revolution#15272]
### Summary
When trying to access the "Resource Groups" page in manager an "Access denied" message appears.

From looking at the code the permissions `resourcegroup_resource_list` and `resourcegroup_resource_edit` are required on the `mgr` context to access the page.

The user in question has both permissions on this context.

However, as it turns out, the `modAccessibleObject::checkPolicy()` method expects the user not only to have those permissions, but also to have them as part of a single policy.

In other words, in `manager/controllers/default/security/resourcegroup/index.class.php` this code (actual code)

```php
return $this->modx->hasPermission(array('resourcegroup_resource_list' => true,'resourcegroup_resource_edit' => true));
```

and this code (example)

```php
return $this->modx->hasPermission('resourcegroup_resource_list') && $this->modx->hasPermission('resourcegroup_resource_edit');
```
would produce different results.

### Background

In our setup the permissions are separated by multiple levels. So there is one policy which includes the `resourcegroup_resource_list` permission and another policy which includes the `resourcegroup_resource_edit` permission. Both policies are assigned to a single user group and multiple groups are assigned to users based on their tasks.

### Expected behavior

Assuming there are two policies:
```
policy p1 {
  resourcegroup_resource_list
}

policy p2 {
  resourcegroup_resource_edit
}
```
and two user groups with those policies assigned
```
group g1 {
  context mgr {
    authority 9999 {
      policy = p1
    }
  }
}

group g2 {
  context mgr {
    authority 9999 {
      policy = p2
    }
  }
}
```
and a user who is a member of both groups
```
user u1 {
  groups { 
    g1, authority=9999
    g2, authority=9999
  }
}
```

I would expect the effective permissions of the user to be
```
user u1 {
  context mgr {
    resourcegroup_resource_list authority=9999
    resourcegroup_resource_edit authority=9999
  }
}
```
while currently they are
```
user u1 {
  context mgr {
    resourcegroup_resource_list authority=9999
  }
  context mgr {
    resourcegroup_resource_edit authority=9999
  }
  
}
```

### Environment
MODx Revolution 2.7.2-pl

**Comment by Mark-H** [comment]
I've not yet looked into this, but I must say this is probably one of the best bug reports I've recently seen.

**Comment by opengeek** [comment]
I can absolutely confirm that this is how the system currently works, as it is comparing the array of permissions against each policy and if they do not all match within a single policy, it will fail the check.

**Comment by opengeek** [comment]
Obviously, the easiest solution would be to give the policy that has the `resourcegroup_resource_edit` permission the `resourcegroup_resource_list` permission, as well.

**Comment by Ruslan-Aleev** [comment]
The access settings lack grouping and hierarchy in the settings themselves.
For example, you can accidentally disable the setting - `view` (Basic access to view all xpdo objects (resources, chunks, snippets, plugins)), and at the same time continue to configure, for example, `save_chunk`, although without `view` it no longer makes sense. Although it is more logical to disable / block settings that depend on the `view`.
And here is the same problem.

**Comment by opengeek** [comment]
> The access settings lack grouping and hierarchy in the settings themselves.
> For example, you can accidentally disable the setting - `view` (Basic access to view all xpdo objects (resources, chunks, snippets, plugins)), and at the same time continue to configure, for example, `save_chunk`, although without `view` it no longer makes sense. Although it is more logical to disable / block settings that depend on the `view`.
> And here is the same problem.

I'm not sure how that is relevant to this issue.

**Comment by Ruslan-Aleev** [comment]
> > The access settings lack grouping and hierarchy in the settings themselves.
> > For example, you can accidentally disable the setting - `view` (Basic access to view all xpdo objects (resources, chunks, snippets, plugins)), and at the same time continue to configure, for example, `save_chunk`, although without `view` it no longer makes sense. Although it is more logical to disable / block settings that depend on the `view`.
> > And here is the same problem.
> 
> I'm not sure how that is relevant to this issue.

I may have misunderstood the issue, but is it logical to be able to turn on `resourcegroup_resource_edit` without turning on `resourcegroup_resource_list`? And this is a more global problem in the access settings. And even if we solve a particular problem in resource groups, this does not exclude the occurrence of problems of this kind in other sections.

**Comment by Ruslan-Aleev** [comment]
Sorry I didn't quite understand the issue. Here it is about the condition for checking access, but I thought that the problem is in setting up access (the ability not to select dependent settings).

**Comment by rthrash** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/user-access-levels-user-accounts-assistance/3794/6
